app/controllers/observer_controller/search_controller.rb

```diff
@@ -23,9 +24,16 @@ def pattern_search
     case type
     when :observation, :user
       ctrlr = :observer
-    when :comment, :herbarium, :image, :location,
-      :name, :project, :species_list, :herbarium_record
+    when :comment, :image, :location, :name, :project, :species_list,
+      :herbarium_record
       ctrlr = type
+    when :herbarium
+      redirect_to_search_or_index(
+        pattern: pattern,
+        search_path: herbaria_searches_path(pattern: pattern),
+        index_path: herbaria_alls_path
+      )
+      return
```

Comment:
As more of our legacy controllers get updated, these lines should be generalized. I'd prefer to wait on that. 
**Question: Should I instead generalize now, based on the one example?**

Reply:
I do like the idea of generalizing, if it can be done. The action seems pretty standard.

Reply:
Thanks for the input @nimmolo.
I will re-think this (both generalization and whether to route searches to #index.